src/compiler/checker.ts

```diff
@@ -34554,8 +34554,11 @@ namespace ts {
                 : isPropertyAccessExpression(location) ? location.name
                 : isBinaryExpression(location) && isIdentifier(location.right) ? location.right
                 : undefined;
+            const isPropertyExpressionCast = isPropertyAccessExpression(location)
+                && isParenthesizedExpression(location.expression)
+                && isAssertionExpression(location.expression.expression);

```

Comment:
It is weird that adding a layer of parens will throw this off. Maybe use `skipParentheses`

Reply:
Good idea